File: liqp-core/src/main/java/liqp/filters/Strip.java
package liqp.filters;

import liqp.nodes.RenderContext;

public class Strip extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        if (!super.isString(value)) {
            return value;
        }

        return super.asString(value).trim();
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Strip.kt
package liqp.filter

import liqp.context.LContext
import liqp.params.FilterParams

class Strip : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return when(value) {
      is String-> value.trim()
      else-> value
    }
  }
}
